### PR TITLE
Add parameter n_jobs for multiprocessing to hyperopt_estimator

### PR DESCRIPTION
**Pull request by DavidBreuer**
Hi, this PR addresses the discussion in issue https://github.com/hyperopt/hyperopt-sklearn/issues/82 and especially comment https://github.com/hyperopt/hyperopt-sklearn/issues/82#issuecomment-430963445. 

To support (at least some) multiprocessing, I added the well-known sklearn/joblib parameter n_jobs to the hyperopt_estimator function. Whenever an estimator is called which supports multiprocessing, it is passed n_jobs as argument:

`estim = hpsklearn.HyperoptEstimator(..., n_jobs=2)`

Two caveats: First, for smaller data sets, this rather slows some sklearn functions down due to parallelization overhead (cf. e.g. https://github.com/scikit-learn/scikit-learn/issues/6645 or https://github.com/scikit-learn/scikit-learn/issues/8216). However, a quick analysis showed that for larger datasets there may be some benefit by using multiple cores: 

Figure_1

Legend: Color encodes number of samples for a dummy sklearn.datasets.make_classification task. y-axis shows time needed to perform 30 evals relative to time with one core. Random seeds were set to ensure that hyperopt runs were identical. Repetitions would be needed for reliable time estimates etc. but this should be enough for demonstration purposes.

Second, it would be nice to parallelize the cross-validation part in _cost_fn but there is an interesting for-else statement that I could not handle using joblib... :) Suggestions are welcome.

I'm happy for ideas how to improve this PR. In case this PR should not be merged, that's perfectly fine, too. Thanks and best!

### CODE REVIEW

**Comment by linminhtoo** [comment]
I support this PR. May I ask any reason for it hasn't been merged?

**Comment by DavidBreuer** [comment]
Thanks for the support. I rebased my branch with the current original master branch to re-enable merging. If further adjustments are needed, please let me know.

File: hpsklearn/estimator.py [hunk 1]
```diff
@@ -525,6 +527,12 @@ def __init__(self,
         self.fit_increment_dump_filename = fit_increment_dump_filename
         self.use_partial_fit = use_partial_fit
         self.refit = refit
+        self._best_preprocs = ()
+        self._best_ex_preprocs = ()
+        self._best_learner = None
+        self._best_loss = None
+        self._best_iters = None
+	self.n_jobs = n_jobs
```
Comment (bjkomer) [inline hpsklearn/estimator.py:535]: The spacing needs to be fixed here. Also, I don't see n_jobs given as a parameter to this function, possibly something got lost in the rebase?

**Comment by bjkomer** [comment]
Thanks for the PR! I missed this earlier, but will take a look through this weekend and test it out.

**Comment by bjkomer** [comment]
Looks good to me!
